package.json

```diff
@@ -4,7 +4,11 @@
   "description": "Webpack Encore is a simpler way to integrate Webpack into your application",
   "main": "index.js",
   "scripts": {
-    "test": "mocha --reporter spec test --recursive",
+    "test": "run-s test:*",
+    "test:normal": "mocha --reporter spec test --recursive --ignore test/persistent-cache/*",
+    "test:persistent-cache-clear": "cross-env CLEAR_PERSISTENT_CACHE=true npm run test:persistent-cache:base",
+    "test:persistent-cache": "npm run test:persistent-cache:base",
```

Comment:
I'd use `yarn run` rather than `npm run` given that we use yarn

Reply:
I totally agree, I only used npm because it was already used in the `travis:lint` command. Should I adjust them all?

Reply:
So we want to run `test/persistent-cache` twice? Once with `CLEAR_PERSISTENT_CACHE` and once without? Testing it both ways catches different behavior?

Would it be possible to, in `test/persistent-cache/functional.js`, loop over each test and run it twice each time? Or, purposely execute Webpack 2x for each test case: 

A) clear the cache
B) Execute webpack once (assert a few things are ok from a fresh cache)
C) Execute webpack again (assert a few things are ok from a NOT fresh cache)

I would love to get rid of the `cross-env`

Reply:
Actually, I tried that, and it didn't work :( 

The webpack cache is only written to disk (and thus available for a new invocation) when all tests have run (the mocha invocation is done) and the node process is exiting. I'm not sure why this is the case, but I couldn't figure out a way to work around this limitation. With my first iteration I actually tried running the webpack invocation twice by adding a `runTwice` parameter to the [`runWebpack`](https://github.com/symfony/webpack-encore/blob/972c8827500f3095dbaff06895a47ac0621bf28e/test/helpers/setup.js#L62) helper and so on. This was the only way I could get it to work reliably.

The order of the tests and cache creation are important, due to the persistent cache: originally, the cache files are always cleared when [the functional tests are started](https://github.com/symfony/webpack-encore/blob/972c8827500f3095dbaff06895a47ac0621bf28e/test/helpers/setup.js#L69). This is unwanted behaviour for the persistent cache tests (remember, I need to finish the node process to flush the webpack cache to disk), which is why I moved those tests into a separate test file. However, clearing the cache is still important for the first invocation, as we want that to be run without cache, which is why I added the ENV switch and the clear separation in scripts for transparency.

So, the final order becomes something like this:
  - `yarn run test`
    - `test:normal` is started
      - Whole cache folder (`test_tmp`) is cleared, which includes the webpack cache
      - All tests, except for the persistent cache tests, are run
      - No webpack cache is written, as there are no tests with persistent cache in this run
    - `test:persistent-cache-clear` is started
      - Whole cache folder (`test_tmp`) is cleared, which includes the webpack cache
      - Persistent cache tests are run, but without cache
      - Webpack cache is flushed to disk
    - `test:persistent-cache` is started
      - No cache is cleared, so the webpack cache now is available
      - Persistent cache tests are run, with cache
      - Updated cache is flushed to disk

Possible solutions:
  - We could rely on the cache clear from the earlier run normal tests: it works, but it not very transparent.
  - Find another crossplatform way of signaling the test that it should clear the cache on invocation. Running another script would also be fine with me by the way. 
  - Maybe the creation of a dedicated test file could help here, something like below. This removed the need of both new dev dependencies, but I'm not sure if this plays fair with all tooling.
```js
const { execSync } = require('child_process');
const { emptyTmpDir } = require('./test/helpers/setup');

execSync('mocha --reporter spec test --recursive --ignore test/persistent-cache/*', { stdio: 'inherit' });

emptyTmpDir();
for (let i = 0; i < 2; i++) {
    execSync('mocha --reporter spec test/persistent-cache --recursive', { stdio: 'inherit' });
}
```